src/NuGet.Services.AzureSearch/VersionList/MutableIndexChanges.cs

```diff
@@ -13,22 +14,86 @@ namespace NuGet.Services.AzureSearch
     /// </summary>
     internal class MutableIndexChanges
     {
+        internal static readonly IReadOnlyList<SearchFilters> AllSearchFilters = Enum
+            .GetValues(typeof(SearchFilters))
+            .Cast<SearchFilters>()
+            .ToList();
+
+        /// <summary>
+        /// This is a dictionary where the key is the state transition. The key is itself a
+        /// <see cref="KeyValuePair{TKey, TValue}"/> where the key is the existing state and the value is the added
+        /// state. The value of the dictionary is the resulting state from that transition. Remember that versions
+        /// are processed in descending version order so some state transition should not be possible.
+        /// </summary>
+        private static readonly IReadOnlyDictionary<KeyValuePair<SICT, SICT>, SICT> AcceptableTransitions
```

Comment:
Can we have domain-specific alternative to `KeyValuePair`? Its property names does not provide any context...

Reply:
Yeah, I think I was lazy at that moment. I'll see what I can do.